### PR TITLE
fix: make compatible with Weather Blocker

### LINKED ISSUE #130: FXMaster V2.0.1 is incompatible with Weather Blocker
### Expected Behavior

Using both FXMaster and Westher Blocker works fine, blocking from Westher Blocker masks FXMaster weather.

### Current Behavior

When using both FXMaster and Weather Blocker in a scene with a background image, a black square hides the scene.

### Steps to Reproduce

1. Create a scene with a background image
2. Activate both FXMaster and Weather Blocker
3. See black scene

### Context

Reported on discord: https://discord.com/channels/170995199584108546/722566083499786352/924316017730150400

The problem is that Weather Blocker also sets a mask on the `fxmaster`, resulting in FXMaster's own mask becoming a regular Graphics object that is rendered.

### Version

v2.0.1

### Foundry VTT Version

V9.238

### Operating System

All

### Browser / App

Native Electron App, Chrome, Firefox, Microsoft Edge, Safari

### Game System

dnd5e, but is irrelevant

### Relevant Modules

weatherblock

### PR DESCRIPTION
**Pull request by ghost91-**
Closes #130

### CODE REVIEW

File: src/module/weatherEffects/WeatherLayer.js [hunk 1]
```diff
@@ -114,23 +135,34 @@ export class WeatherLayer extends CanvasLayer {
   async tearDown() {
     Object.values(this.weatherEffects).forEach(({ fx }) => fx.stop());
 
-    this.weather = null;
+    this.weather = undefined;
     this.mask = null;
```
Comment (ghost91-) [inline src/module/weatherEffects/WeatherLayer.js:139]: @dev7355608 Any idea what might cause the scene to crash when it is redrawn (e.g. due to the background image being changed) when this line is removed and PV is active? Maybe related to this change? https://github.com/dev7355608/perfect-vision/commit/5a1452d3184fdd38630cc21eebbc9ca83197decf

Comment (dev7355608) [reply src/module/weatherEffects/WeatherLayer.js]: Tear down destroys all objects, but doesn't clear the mask field of the layer. So once rendering continues it tries to use the destroyed mask, which fails horribly.

Comment (ghost91-) [reply src/module/weatherEffects/WeatherLayer.js]: But I never set a mask myself. I see that a mask is being set, probably by PV. But then it should also be PVs responsibility to set it back to `null` after it being destroyed.

Comment (ghost91-) [reply src/module/weatherEffects/WeatherLayer.js]: The cleanup of the mask will added to PV.

**Pull request merged**